### PR TITLE
feat(testing): update jest-preset-angular to v8.0.0

### LINKED ISSUE #1979: jest-preset-angular v8 has breaking changes
_[Please make sure you have read the submission guidelines before posting an issue](https://github.com/nrwl/nx/blob/master/CONTRIBUTING.md#-submitting-issue)_

# Prerequisites

Please answer the following questions for yourself before submitting an issue.
**YOU MAY DELETE THE PREREQUISITES SECTION.**

- [x] I am running the latest version
- [x] I checked the documentation and found no answer
- [x] I checked to make sure that this issue has not already been filed
- [x] I'm reporting the issue to the correct repository (not related to Angular, AngularCLI or any dependency)

## Expected Behavior

Running jest tests does not fail

## Current Behavior

Jest tests fail with the following error message:

```
Module jest-preset-angular/AngularSnapshotSerializer.js in the snapshotSerializers option was not found.
```

### Steps to Reproduce

1. update jest-preset-angular to 8.0.0
2. run jest tests

## Other

I looked into jest-preset-angular and it seems that they introduced a build folder with v8 thus the path is not correct anymore.

### PR DESCRIPTION
**Pull request by mehrad-rafigh**
First of all, happy new year to all of you,

This PR supersedes #1986 due to inactivity and closes #1979 and #2165 
I added the new snapshot serializer that was mentioned by @wtho and updated everything accordingly. 

Hoping we can merge this asap :)

### CODE REVIEW

**Review by overlineink — APPROVED** [review]
LGTM! Thanks 💙, @mehrad-rafigh. I hope this PR been merged as soon as possible.

File: packages/jest/src/migrations/update-8-10-0/update-8-10-0.ts
Comment (FrozenPandaz) [inline packages/jest/src/migrations/update-8-10-0/update-8-10-0.ts]: We should go through all of the architect targets.. for targets which use `@nrwl/jest:jest` as the builder... and use the `jestConfig` option verbatim if it exists so that we can catch all `jest.config.js` which are referenced in `workspace.json`.

File: packages/jest/src/migrations/update-8-10-0/update-8-10-0.ts
Comment (FrozenPandaz) [inline packages/jest/src/migrations/update-8-10-0/update-8-10-0.ts]: Awesome work here!

Unfortunately, this didn't work quite as expected for me. This is the resulting file:

```
module.exports = {
  name: 'products',
  preset: '../../jest.config.js',
  coverageDirectory: '../../coverage/apps/products',
  snapshotSerializers: [
    jest-preset-angular/build/AngularNoNgAttributesSnapshotSerializer.js'jest-preset-angular/build/AngularSnapshotSerializer.js',
    'jest-preset-angular/build/HTMLCommentSerializer.js'
  ]
};

```

**Review by FrozenPandaz — CHANGES_REQUESTED** [review]
Thank you for your contribution!

Can you please keep the commit by @JoshMentzer so his work is attributed to him?

File: packages/jest/src/migrations/update-8-10-0/update-8-10-0.ts
Comment (mehrad-rafigh) [reply packages/jest/src/migrations/update-8-10-0/update-8-10-0.ts]: Hi @FrozenPandaz. Thanks for the review and the bug catch! 
I will update the PR asap. As for the commit authors, I can add @JoshMentzer as a co-author. I just need to find out his github email address :)

File: packages/jest/src/migrations/update-8-10-0/update-8-10-0.ts
Comment (mehrad-rafigh) [reply packages/jest/src/migrations/update-8-10-0/update-8-10-0.ts]: I don't get your point. It something not correct here?

File: packages/jest/src/migrations/update-8-10-0/update-8-10-0.ts
Comment (mehrad-rafigh) [reply packages/jest/src/migrations/update-8-10-0/update-8-10-0.ts]: I updated my PR to resolve this issue.

**Comment by mehrad-rafigh** [comment]
Hi @FrozenPandaz. Would you please review the changes I made? We really need this in nx

**Comment by mehrad-rafigh** [comment]
@FrozenPandaz pinging you again, hoping you will review this :)

File: CONTRIBUTING.md
Comment (FrozenPandaz) [inline CONTRIBUTING.md]: Thanks for fixing this typo but it's a separate change. In the future, please make a separate PR for separate changes.

File: packages/jest/src/migrations/update-8-10-0/update-8-10-0.ts
Comment (FrozenPandaz) [reply packages/jest/src/migrations/update-8-10-0/update-8-10-0.ts]: Sorry, let me clarify, this only migrates targets which have a `test` target that uses `@nrwl/jest:jest`.

A user could have the following:

```
projects: {
  app1: {
    architect: {
      e2e: {
        builder: '@nrwl/jest:jest',
        options: { ... }
      }
    }
  }
```

Such projects should be migrated as well.

**Review by FrozenPandaz — CHANGES_REQUESTED** [review]
I don't think having him as a co-author of your commit will persist after someone merges this PR because it assigns the merger as the co-author?

Can you have his commit there separately and we won't squash when we merge? This likely involves remaking your branch.. cherry-picking his PR and then adding your changes on top as a separate commit...

File: CONTRIBUTING.md
Comment (mehrad-rafigh) [reply CONTRIBUTING.md]: You're right. I will take it back and open a new PR

**Comment by mehrad-rafigh** [comment]
I will take a look tonight. If I can cherry pick the commit without too much hustle, I will do it.
If its too much of a hustle I will close this PR and somebody else can give it a try.

**Comment by wtho** [comment]
Sorry to mess with the discussion in this PR, but it is no big deal to incorporate the original commit.
Cherry-picking takes the file changes, but overrides the committer in the metadata. To incorporate the original committer, follow these steps (there might be more elegant ways, but I like to work file-based, when there are issues with git):
* Checkout the original commit's repo to a new folder (https://github.com/JoshMentzer/nx/tree/feat(testing)-update-jest-preset-angular). You will then have his commit as the last one.
* Copy your *changed files* to that newly checked out folder and override whatever was there before.  To see which files these are, you can just check [here](https://github.com/nrwl/nx/pull/2250/files) in this PR. Then make a single commit with your changes. Set the git origin to your repo, change the branch to this one and force-push.

Even if you do not want to go on with this PR anymore, this way someone else can continue with your work, and you will still author your work in your second commit.
Anyway I think you should go on, you are almost done!

Cheers and happy coding!

**Pull request closed without merging**
